snooty/test_types.py

```diff
@@ -23,6 +23,7 @@ def test_project() -> None:
     project_config, project_diagnostics = ProjectConfig.open(Path(".").resolve())
     assert project_config.name == "unnamed"
     assert project_config.title == "untitled"
+    assert project_config.deprecated_versions == {}
```

Comment:
Since we have a fairly limited use case for `deprecated_versions` data (only in docs-landing), I think I would prefer to not write that field to the metadata collection at all if nothing is specified. Do you think that makes sense?